### PR TITLE
feat: kosmosctl join and unjoin

### PR DESCRIPTION
**Pull request by rxy0210**

### CODE REVIEW

File: pkg/kosmosctl/join/join.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
+package join
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/kosmos.io/kosmos/pkg/kosmosctl/manifest"
```
Comment (ONE7live) [inline pkg/kosmosctl/join/join.go]: Pay attention to specifications：
system package
blank line
Third party package
blank line
The package of the current project

File: pkg/kosmosctl/manifest/manifest_rbac.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package manifest
```
Comment (ONE7live) [inline pkg/kosmosctl/manifest/manifest_rbac.go]: Maybe we can classify resources?

File: pkg/kosmosctl/manifest/manifest_operator.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                  - key: app
+                    operator: In
+                    values:
+                      - operator
+              namespaces:
+                - clusterlink-system
+              topologyKey: kubernetes.io/hostname
+      containers:
+      - name: operator
+        image: ghcr.io/kosmos-io/clusterlink/clusterlink-operator:__VERSION__
```
Comment (ONE7live) [inline pkg/kosmosctl/manifest/manifest_operator.go]: change into: ghcr.io/kosmos-io/clusterlink-operator:

Comment (rxy0210) [reply pkg/kosmosctl/manifest/manifest_operator.go]: ok

File: pkg/kosmosctl/join/join.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	Client        kubernetes.Interface
+}
+
+// NewCmdJoin join this to Clusterlink control plane
+func NewCmdJoin(f ctlutil.Factory) *cobra.Command {
+	o := &CommandJoinOptions{}
+
+	cmd := &cobra.Command{
+		Use:                   "join",
+		Short:                 i18n.T("Join this to clusterlink control plane"),
```
Comment (duanmengkk) [inline pkg/kosmosctl/join/join.go:53]: what 18n means? seems so strange

Comment (ONE7live) [reply pkg/kosmosctl/join/join.go:53]: Quoted from kubectl, internationalization/i18n？the same as kubernetes/k8s

Comment (rxy0210) [reply pkg/kosmosctl/join/join.go:53]: 18n is an abbreviation for internationalization, and it shows different languages depending on locale, as I've seen used in other codes

File: pkg/kosmosctl/unjoin/unjoin.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+	err := o.DynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "kosmos.io",
+		Version:  "v1alpha1",
+		Resource: "clusters",
+	}).Namespace("").Delete(context.TODO(), args[0], metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("(cluster) kosmosctl unjoin run error, delete cluster failed: %s", err)
+	}
+
+	for {
```
Comment (hanweisen) [inline pkg/kosmosctl/unjoin/unjoin.go:120]: we should wait some time after recheck

Comment (rxy0210) [reply pkg/kosmosctl/unjoin/unjoin.go:120]: time.Sleep(3 * time.Second)

Comment (hanweisen) [reply pkg/kosmosctl/unjoin/unjoin.go:120]: 👌

File: pkg/kosmosctl/unjoin/unjoin.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+			Group:    "kosmos.io",
+			Version:  "v1alpha1",
+			Resource: "clusters",
+		}).Namespace("").Get(context.TODO(), args[0], metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				break
+			}
+			return fmt.Errorf("(cluster) kosmosctl unjoin run error, delete cluster failed: %s", err)
+		}
```
Comment (hanweisen) [inline pkg/kosmosctl/unjoin/unjoin.go:131]: put time.Sleep(3 * time.Second)  here may be better

Comment (rxy0210) [reply pkg/kosmosctl/unjoin/unjoin.go:131]: why not first wait 3 second. When I test delete cluster, I usually can't delete it right away.need wait time

Comment (hanweisen) [reply pkg/kosmosctl/unjoin/unjoin.go:131]: you can refer https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/
some controller will add finalizer on cluster, let cluster can not delete immediately

Comment (rxy0210) [reply pkg/kosmosctl/unjoin/unjoin.go:131]: because cluster have finalizer, i think that we should wait before the first get.

**Pull request merged**